reduction/lr_reduction/event_reduction.py

```diff
@@ -326,17 +329,25 @@ def extract_meta_data_4B(self):
         else:
             self.det_distance = self.DEFAULT_4B_SAMPLE_DET_DISTANCE
 
-        if "source-det-distance" in settings:
+        if self.use_emission_time:
```

Comment:
Good point. I added a check during meta data extraction. If the logs are not there, then we turn the calculation off and use an effective distance instead (the old way of doing it, which works too).

Reply:
Good catch. Making that check makes a system test fail, because that data is older and doesn't have that log. That would create silly results. I'll change the test accordingly.

Reply:
@backmari done